pennylane/drawer/tape_mpl.py

```diff
@@ -251,7 +254,7 @@ def _tape_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, *, fig
         decimals=decimals,
         active_wire_notches=active_wire_notches,
         bit_map=bit_map,
-        terminal_layers=[cl[-1] for cl in cwire_layers],
+        terminal_layers=[cl[-1][-1] for cl in cwire_layers.values()],
```

Comment:
`cwire_layers.values()` is a "dict values" of list of lists, instead of the previous list of lists. So we have one indentation level more, in which we also pick the last entry.